modules/KalturaSupport/components/offline.js

```diff
@@ -0,0 +1,160 @@
+( function( mw, $ ) {"use strict";
+
+	mw.PluginManager.add( 'offline', mw.KBasePlugin.extend({
+		defaultConfig: {
+
+		},
+		setup: function(){
+			this.bind('playerReady', function() {
+			var _XMLHttpRequest = XMLHttpRequest;
+			window.requestFileSystem  = window.requestFileSystem || window.webkitRequestFileSystem;
+			window.requestFileSystem(window.TEMPORARY, 20*1024*1024 /*5MB*/, onInitFs, errorHandler);
```

Comment:
deprecated, use 
var requestedBytes = 1024*1024*5;
navigator.webkitPersistentStorage.requestQuota (
        requestedBytes, function(grantedBytes) {
            console.log('we were granted ', grantedBytes, 'bytes');
            window.requestFileSystem(PERSISTENT, grantedBytes, onInitFs, errorHandler);
        }, function(e) { console.log('Error', e); }
);